### PR TITLE
Revert #7677

### PR DESCRIPTION
**Pull request by afshin**
Adds `--max-old-space` script back to `buildutils` to support Node v10+

## References
https://github.com/jupyterlab/jupyterlab/issues/8083
https://github.com/jupyterlab/jupyterlab/pull/7677/

## Code changes
Reverts #7677

## User-facing changes
N/A

## Backwards-incompatible changes
N/A

### CODE REVIEW

**Comment by blink1073** [comment]
A `package-lock.json` snuck in there, whoops!

**Comment by blink1073** [comment]
Can you please also update our [engines](https://github.com/jupyterlab/jupyterlab/blob/master/dev_mode/package.json#L187) spec for node 10+?  We should point to this file in [CONTRIBUTING.md](https://github.com/jupyterlab/jupyterlab/blob/master/CONTRIBUTING.md#installing-nodejs-and-jlpm) to keep them in sync and as a reminder of where the value is enforced.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ You can launch a binder with the latest JupyterLab master to test something (thi
 
 ### Installing Node.js and jlpm
 
-Building JupyterLab from its GitHub source code requires Node.js. The development version requires Node.js version 12+.
+Building JupyterLab from its GitHub source code requires Node.js. The development version requires Node.js version 10+.
```
Comment (blink1073) [inline CONTRIBUTING.md]: Can you please add a note that this is defined in ` dev_mode/package.json:engines`?  I had to look in commands.py to see how we were enforcing the node version.

**Comment by afshin** [comment]
Fixes #8083

**Pull request merged**